### ISSUE TITLE
Enable searching Home Manager options from search.nixos.org

### ISSUE THREAD

**Issue by berbiche** [nix-community/home-manager#2297]
### Description

With the merge of this PR: <https://github.com/NixOS/nixos-search/pull/324>, it seems like it would be possible to search Home Manager options from <https://search.nixos.org/flakes>.

What would the Home Manager project need to do @ysndr @ncfavier :smiley:?

edit: link: https://github.com/NixOS/nixos-search/issues/323

**Comment by ncfavier** [comment]
Would love to see this. I think the main problem is that there isn't an official flake output for "modules that are not nixos- or darwin-specific", like home-manager modules. Maybe we could just start using `modules` unofficially though.

**Comment by ysndr** [comment]
Exactly, the main problem right now is that the modules hm provides are not publicated by the flake
There’s the integration with the nixos configuration but that’s all as far as I can see.
I don’t know if there’s big hurdles to make hm’s config nixos compatible (it shouldn’t as it already integrates) and make them publicly visible as a module in the flake

**Comment by terlar** [comment]
Still relevant. Docs is already part of the flake now. Not sure if it is in the correct format to be picked up though.

**Comment by ncfavier** [comment]
For the record I just looked into this a bit, and realised that exposing the modules is not enough: we would also need to expose the `lib` that HM modules expect (`stdlib-extended.nix`) in order to be able to evaluate them.

Another option might be to expose *options* (the argument to `lib.optionAttrSetToDocList`, probably) directly in the flake rather than modules. That way nixos-search doesn't need to know about the specifics of how HM modules expect to be evaluated (and we might imagine adding support for other "module scopes" in the future).

On the nixos-search side, it should be rather straightforward to add a `scope` field to options (like `nixos`, `home-manager`, etc.) and display different scopes in separate tabs.

**Comment by eveeifyeve** [comment]
@rycee this issue can be closed.

**Comment by teto** [comment]
home-manager is now available as a source eg: https://search.nixos.org/options?channel=unstable&query=neovim&source=home_manager
